dopamine/discrete_domains/run_experiment.py

```diff
@@ -607,37 +622,58 @@ def _run_experiment_loop(self):
     iterations is run.
     """
     threads = []
+    train_iterations = threading.Semaphore(self._max_simultaneous_iterations)
+    eval_iterations = threading.Semaphore(1)
     self._completed_iteration = self._start_iteration
+
     for iteration in range(self._start_iteration, self._num_iterations):
-      self._running_iterations.acquire()
-      threads.append(self._run_one_iteration(iteration))
+      if (iteration + 1) % self._eval_period == 0:
```

Comment:
`threaded_method` was removed in master. So this now looks different after the merge. Would you mind taking another look and see if this still requires a merge?

Also we have as a TODO to revamp this `run_iterations` method.

Reply:
This is a whole lot easier for me to read, now.  Thanks for integrating from master.

I do still believe that this gets clearer, still, when `_run_one_iteration` is split into `_run_training_iteration` and `_run_eval_iteration`.  What are your thoughts about that?

Reply:
I agree with you, but at the same time, don't like splitting like this because it leads to a lot of repeated code.

I think this might change in the revamp (where we get rid of the internal iteration counter, and store checkpoints differently).

I prefer to leave as is for now to avoid extra-work. Ok?

Reply:
Totally.  I think the refactor we talked about earlier would align both eval and train cases, too.  Leaving it as-is seems like a good decision to me.